### PR TITLE
C extension PHP 5.4 support

### PR DESCRIPTION
**Pull request by pawelchcki**
### Readiness checklist
- [ ] [Changelog entry](docs/changelog.md) added, if necessary
- [ ] Tests added for this feature/bug

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -27,6 +27,8 @@ x-aliases:
 
 services:
   '5.4-debug': { <<: *base_php_service, build: 'dockerfiles/debug_php-5.4' }
+  '5.4': { <<: *base_php_service, build: 'dockerfiles/php-5.4' }
```
[the remark this answers is not in the document]
Comment (pawelchcki) [reply docker-compose.yml]: We should :D 
And its been done !

File: src/ext/dispatch.c [hunk 2]
```diff
@@ -113,13 +117,19 @@ static void execute_fcall(ddtrace_dispatch_t *dispatch, zend_execute_data *execu
         }
         goto _exit_cleanup;
     }
+       if (return_value_ptr && *return_value_ptr) {
```
[the remark this answers is not in the document]
Comment (pawelchcki) [reply src/ext/dispatch.c]: A ton of bad indentations - 've run clang-format on the code. Now it should be much nicer

File: src/ext/dispatch.c [hunk 3]
```diff
@@ -161,27 +171,82 @@ static zend_always_inline zend_bool executing_method(zend_execute_data *execute_
 #endif
 }
 
+
+void zend_throw_exception_internal(zval *exception TSRMLS_DC) /* {{{ */
+{
+    DD_PRINTF("Hmm");
```
[the remark this answers is not in the document]
Comment (pawelchcki) [reply src/ext/dispatch.c]: Removed

File: src/DDTrace/Integrations/Mongo/MongoClientIntegration.php [hunk 4]
```diff
@@ -6,13 +6,18 @@
...
 use DDTrace\Integrations\Integration;
+use DDTrace\Util\Environment;
 
 final class MongoClientIntegration extends Integration
 {
     const CLASS_NAME = 'MongoClient';
 
     protected static function loadIntegration()
     {
+        if (Environment::matchesPhpVersion('5.4')) {
```
Comment (pawelchcki) [inline src/DDTrace/Integrations/Mongo/MongoClientIntegration.php]: 👍

**Comment by SammyK** [comment]
Awesome work on this @pawelchcki! I know this was a beast and you totally rocked it! :D

**Pull request merged**